stdlib/3.7/dataclasses.pyi

```diff
@@ -0,0 +1,53 @@
+from typing import overload, Any, Callable, Dict, Generic, Iterable, Mapping, Optional, Tuple, Type, TypeVar, Union
+
+
+_T = TypeVar('_T')
+
+_DictType = TypeVar('_DictType', bound=dict)
+_TupleType = TypeVar('_TupleType', bound=tuple)
+
+class _MISSING_TYPE: ...
+
+class _InitVarMeta(type): ...
+
+def asdict(obj: Any, *, dict_factory: _DictType = ...) -> _DictType: ...
```

Comment:
This is not quite correct; the argument type should be something like `Type[_DictType]`.

Also, the range of acceptable type is a little wider than what's given here: in practice, dict_factory can be any callable accepting a list of (key, value) tuples. We could perhaps better type this as: `def asdict(obj: Any, *, dict_factory: Callable[[List[Tuple[str, Any]], _T]) -> _T: ...`, where `_T = TypeVar('_T')`.

All this also goes for `astuple`.

Reply:
This took me a couple of readings but I now understand and agree that using `Callable` is the most accurate approach.